python/src/main/processing/clean_duplicates.py

```diff
@@ -0,0 +1,100 @@
+"""
+This script allows the user to find out actual full names of GitHub repositories
+to avoid duplicated projects in a dataset.
+The script makes requests to GitHub API, saves unique full names of repositories
+and optionally saves all JSON responses containing repositories metadata.
+It accepts
+    * path to csv file --  dataset, f.e. downloaded from https://seart-ghs.si.usi.ch/
+    * path to output directory
+    * optional parameter to save metadata
+    * index to start from (default 0)
+"""
+
+import json
+import os
+from typing import Set, TextIO, Tuple
+
+import pandas as pd
+import argparse
+import requests
+import logging
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+from python.src.main.utils.file_utils import create_directory
+
+
+def main():
+    logging.basicConfig(level=logging.DEBUG)
+    args = parse_args()
+    jsons_dir_path, results_path = create_output_paths(args.output, args.save_metadata)
+
+    dataset = pd.read_csv(args.csv_path)
+    os.environ['GIT_TERMINAL_PROMPT'] = '0'
+    token = os.getenv('GITHUB_TOKEN')
+    headers = {'Authorization': f'token {token}'}
+
+    # retries to avoid github api rate limit (403 status code)
+    s = requests.Session()
+    retries = Retry(total=50, backoff_factor=10, status_forcelist=[403])
+    s.mount('https://', HTTPAdapter(max_retries=retries))
+
+    with open(results_path, 'a') as results_fout:
+        if args.start_from == 0:
+            results_fout.write('full_name\n')
+        unique_names = set()
+        for index, project in enumerate(dataset.name[args.start_from:]):
+            username, project_name = project.split('/')
+            query_url = f'https://api.github.com/repos/{username}/{project_name}'
```

Comment:
Btw, there is a library for interaction with GitHub API: [PyGithub](https://github.com/PyGithub/PyGithub) but I haven't used it yet. Maybe we can check it and rewrite the code to use it in another PR?

Reply:
It is a great idea, thank you!